### ISSUE TITLE
window is not defined

### ISSUE THREAD

**Issue by anode7** [jesusbotella/ngx-lazy-load-images#6]
When I install this script in a server / client environment I have this error:
ReferenceError: window is not defined

**Comment by dottodot** [comment]
Sounds like your using Universal rendering. I've just had a play with modifying it so ngOnInit looks like

```
  ngOnInit() {
    if (this.ps.isBrowser) {
      require('intersection-observer');
      this.ngZone.runOutsideAngular(() => this.init());
    }
  }
```
which seems to work

It would also need something to handle the server side request so that the src is set for seo purposes, but at the moment I'm handling that in my own image component.

**Comment by jesusbotella** [comment]
Yes, it seems like the universal rendering is failing because of the Intersection Observer polyfill as @dottodot stated.

@dottodot Do you want to make a PR about this too? It seems like you got it right :)

**Comment by anode7** [comment]
I managed to make it cohabit with the SSR by importing the module in my browser-app.module.ts and not in the app.module.ts
├── app
 │   ├── app.component.ts
 │   ├── app.browser.module.ts
 │   └── app.server.modue.ts
.......
 ├── index.html
 ├── main.browser.ts
 └── main.server.ts

**Comment by dottodot** [comment]
@anode7 That will work in some cases but you may have issues on lazy loaded modules where you might have to use shared module. I'm just sorting a PR for it anyway.

**Comment by anode7** [comment]
In my component :
```
import { Component, OnInit, OnDestroy, Inject, PLATFORM_ID } from '@angular/core';
import {isPlatformBrowser } from '@angular/common';
...
@Component({ ...
})
export class HomeView implements OnInit, OnDestroy {
...
constructor(
    @Inject(PLATFORM_ID) platformId: string,
...
) {
    this.testBrowser = isPlatformBrowser(platformId);//Boolean
}
```

**Comment by jesusbotella** [comment]
@anode7 @dottodot Could you please test the new release?

I included the @dottodot's fix for Angular Universal compatibility

**Comment by dottodot** [comment]
@jesusbotella Apologies for not getting round to doing a PR just got sidetracked with everything I have on. Anyway just tested with Universal SSR and it's working well.

**Comment by anode7** [comment]
@jesusbotella I just tested just now by generating an Angular-cli project with` --service-worker` and `ng generate universal AppName`.
No errors, but images are not displayed.

**Comment by dottodot** [comment]
Are they displayed without SSR?

You may find you need to do what I've done and that's create a component for your images that handles if it's browser or server. So if it browser it's ok to set `data-src` as they will be lazy-loaded, but on the server it needs to set the `src`.

**Comment by anode7** [comment]
@dottodot  yes, without using the SSR the images are displayed
with SSR, images are not displayed on the client

**Comment by dottodot** [comment]
OK so that will be because `ngx-lazy-load-images` doesn't do anything when it server side which is why I said you'll need to setup a component to handle server vs browser images. You don't really want to lazy load ssr images anyway for seo purposes, for for the server side to need to set the src / srcset as normal.

**Comment by jesusbotella** [comment]
@anode7 Could you please send me the example somehow? I want to try what happen to solve the issue.

@anode7 @dottodot Do you think that the plugin should set the src and the srcset attribute to the node if it is rendered in the server bypassing lazy-load?

**Comment by anode7** [comment]
@jesusbotella [testLazyUniversalSw](https://github.com/anode7/testLazyUniversalSw)
example made very quickly without frills

**Comment by jesusbotella** [comment]
As of the new library release, images load as expected when rendered with data attributes coming from SSR.

The issue of images not being rendered in browser when using SSR has been fixed in #9, please go to the issue for more details :)
